apps/rush-lib/src/data/VersionPolicy.ts

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { cloneDeep } from 'lodash';
+import * as semver from 'semver';
+
+import {
+  IVersionPolicyJson,
+  ILockStepVersionJson,
+  IIndividualVersionJson
+} from './RushConfiguration';
+
+import IPackageJson from '../utilities/IPackageJson';
+
+/**
+ * Type of version bumps
+ * @alpha
+ */
+export enum BumpType {
+  'prerelease',
+  'release',
+  'patch',
+  'minor',
+  'major'
+}
+
+/**
+ * Version policy base type names
+ * @alpha
+ */
+export enum BaseTypeName {
+  'lockStepVersion',
+  'individualVersion'
+}
+
+/**
+ * This is the base class for version policy which controls how versions get bumped.
+ * @alpha
+ */
+export abstract class VersionPolicy {
+  private _policyName: string;
+  private _baseType: BaseTypeName;
+
+  public static load(versionPolicyJson: IVersionPolicyJson): VersionPolicy {
+    const baseTypeName: BaseTypeName = BaseTypeName[versionPolicyJson.baseType];
+    if (baseTypeName === BaseTypeName.lockStepVersion) {
+      return new LockStepVersionPolicy(versionPolicyJson as ILockStepVersionJson);
+    } else if (baseTypeName === BaseTypeName.individualVersion) {
+      return new IndividualVersionPolicy(versionPolicyJson as IIndividualVersionJson);
+    }
+    return undefined;
+  }
+
+  constructor(versionPolicyJson: IVersionPolicyJson) {
+    this._policyName = versionPolicyJson.policyName;
+    this._baseType = BaseTypeName[versionPolicyJson.baseType];
+  }
+
+  public get policyName(): string {
+    return this._policyName;
+  }
+
+  public get baseType(): BaseTypeName {
+    return this._baseType;
+  }
+
+  public abstract ensure(project: IPackageJson): IPackageJson | undefined;
+}
+
+/**
+ * This policy indicates all related projects should use the same version.
+ * @alpha
+ */
+export class LockStepVersionPolicy extends VersionPolicy {
```

Comment:
I will leave it as it is while it is in experimental mode.

---
In reply to: [125351027](https://github.com/Microsoft/web-build-tools/pull/259#discussion_r125351027) [](ancestors = 125351027)